packages/iTwinUI-react/src/core/Dialog/DialogTitleBar.tsx

```diff
@@ -48,14 +49,30 @@ export const DialogTitleBar = Object.assign(
       titleText,
       isDismissible = dialogContext.isDismissible,
       onClose = dialogContext.onClose,
+      isDraggable = dialogContext.isDraggable,
       className,
+      onPointerDown: onPointerDownProp,
       ...rest
     } = props;
+
+    const { onPointerDown } = useDialogDragContext();
+    const handlePointerDown = React.useCallback(
+      (event: React.PointerEvent<HTMLDivElement>) => {
+        onPointerDown?.(event);
```

Comment:
Maybe guard this inside a `defaultPrevented` so that the user can cancel this if they want.